tests/Unleash.Tests/Communication/UnleasRequestBodyUnitTest.cs

```diff
@@ -0,0 +1,97 @@
+using NUnit.Framework.Internal;
+using Unleash.Communication;
+using Unleash.Metrics;
+using Unleash.Tests.Mock;
+using Yggdrasil;
+
+namespace Unleash.Tests.Communication
+{
+    public class UnleasRequestBodyUnitTest
+    {
+        private IUnleashApiClient CreateApiClient()
+        {
+            var requestHeaders = new UnleashApiClientRequestHeaders
+            {
+                ConnectionId = "00000000-0000-4000-a000-000000000000",
+            };
+
+            var httpClient = new HttpClient(messageHandler)
+            {
+                BaseAddress = new Uri("http://example.com")
+            };
+            var client = new UnleashApiClient(httpClient, requestHeaders, null);
+            return client;
+        }
+
+        private class MockHttpMessageHandler : HttpMessageHandler
+        {
+            public List<HttpRequestMessage> calls
+            {
+                get;
+                set;
+            } = new List<HttpRequestMessage>();
+
+            protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                calls.Add(request);
+
+                bool isValidConnectionId = false;
+
+                if (request.Content != null)
+                {
+                    string requestBody = await request.Content.ReadAsStringAsync();
+
+                    using (JsonDocument doc = JsonDocument.Parse(requestBody))
+                    {
+                        if (doc.RootElement.TryGetProperty("connectionId", out JsonElement connectionIdElement))
+                        {
+                            if (connectionIdElement.ValueKind == JsonValueKind.String)
+                            {
+                                string connectionId = connectionIdElement.GetString();
+                                isValidConnectionId = connectionId == "00000000-0000-4000-a000-000000000000";
+                            }
+                        }
+                    }
+                }
+
+                return new HttpResponseMessage(isValidConnectionId ? System.Net.HttpStatusCode.OK : System.Net.HttpStatusCode.BadRequest);
+            }
+        }
+
+        private IUnleashApiClient api
+        {
+            get => TestExecutionContext.CurrentContext.CurrentTest.Properties.Get("api") as IUnleashApiClient;
+            set => TestExecutionContext.CurrentContext.CurrentTest.Properties.Set("api", value);
+        }
+
+        MockHttpMessageHandler messageHandler
+        {
+            get => TestExecutionContext.CurrentContext.CurrentTest.Properties.Get("messageHandler") as MockHttpMessageHandler;
+            set => TestExecutionContext.CurrentContext.CurrentTest.Properties.Set("messageHandler", value);
+        }
+
+        [SetUp]
+        public void SetupTest()
+        {
+            messageHandler = new MockHttpMessageHandler();
+        }
+
+
+        [Test]
+        public async Task IdentificationHttpHeaders()
```

Comment:
it's difficult to tell what this test is doing. I'd expect sth like this:
* given a client with some connection id and interval
* when i send metrics or register
* we sent the connection id header and interval header

what we have instead is:
* given a generic client
* when i send metrics or register
* then registration result was boolean true

Reply:
In other words I'd expect to see relevant test data in the test and noise hidden in helper methods. This test hides everything that is relevant (connection id and interval headers checking)

Reply:
Should be better now.